File: Server/src/packets/server/S3EntitySpawnPacket.ts
import EntityType from '../../protocol/entity-type';

export default interface S3EntitySpawnPacket {
  label: string;
  entityId: number;
  x: number;
  y: number;
  z: number;
  rot: number;
  withFocus: boolean;
  withTakeControl: boolean;
  entityType: EntityType;
}

File: Assets/Scripts/Networking/Packets/Server/S3EntitySpawnPacket.cs
using UnityEngine;

public class S3EntitySpawnPacket
{
    public string label { get; set; }
    public int entityId { get; set; }
    public float x { get; set; }
    public float y { get; set; }
    public float z { get; set; }
    public float rot { get; set; }
    public bool withFocus { get; set; }
    public bool withTakeControl { get; set; }
    public EntityType entityType { get; set; }

    public Vector3 ToVector3()
    {
        return new Vector3(x, y, z);
    }
}